roles/boot_iso/tasks/hpe.yml

```diff
@@ -1,24 +1,99 @@
+    command: PowerForceOff
+    baseuri: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
+    username: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
+    password: "{{ hostvars[inventory_hostname]['bmc_password'] }}"
+
+- block:
+    - name: HPE Eject Virtual Media (if any)
+      community.general.redfish_command:
+        category: Manager
+        command: VirtualMediaEject
+        baseuri: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
+        username: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
+        password: "{{ hostvars[inventory_hostname]['bmc_password'] }}"
+        virtual_media:
+          image_url: "{{ boot_iso_url }}"
+        resource_id: 1
+  rescue:
+    - name: Get Virtual Media information
+      community.general.redfish_info:
+        category: Manager
+        command: GetVirtualMedia
+        baseuri: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
+        username: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
+        password: "{{ hostvars[inventory_hostname]['bmc_password'] }}"
+      register: result
+
+    - name: Get blocking virtual_media
+      set_fact:
+        blocking_virtual_media:  "{{ result.redfish_facts.virtual_media.entries
+          | flatten(levels=2)
+          | selectattr('ConnectedVia', 'defined') | list
+          | json_query('[?(
+              ConnectedVia == `URI`
+              && Image != null
+              && (
+                contains(MediaTypes, `CD`)
+                || contains(MediaTypes, `Floppy`)
+                || contains(MediaTypes, `USBStick`)
+                || contains(MediaTypes, `DVD`)
+              )
+            )]'
+          ) | from_yaml
+        }}"
+
+    - name: Attempting to eject blocking media
+      community.general.redfish_command:
+        category: Manager
+        command: VirtualMediaEject
+        baseuri: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
+        username: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
+        password: "{{ hostvars[inventory_hostname]['bmc_password'] }}"
+        virtual_media:
+          image_url: "{{ item.Image }}"
+        resource_id: 1
+      loop: "{{ blocking_virtual_media }}"
+      no_log: True
+
+- block:
+    - name: HPE Insert Virtual Media
+      community.general.redfish_command:
+        category: Manager
+        command: VirtualMediaInsert
+        baseuri: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
+        username: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
         password: "{{ hostvars[inventory_hostname]['bmc_password'] }}"
-        state: "poweroff"
+        virtual_media:
+          image_url: "{{ boot_iso_url }}"
+          media_types:
+            - CD
+        resource_id: 1
 
-    - name: HPE disconnect existing Virtual Media
-      hpilo_boot:
-        host: "{{ hostvars[inventory_hostname]['bmc_address'] }}"
-        login: "{{ hostvars[inventory_hostname]['bmc_user'] }}"
+- block:
+    - name: Set Boot for the HPE
+      uri:
```

Comment:
If you can't use the redfish module can you please add a comment to explain why.